sdk/nodejs/resource.ts

```diff
@@ -327,12 +327,21 @@ export class Output<T> {
     /* @internal */ public readonly promise: () => Promise<T>;
 
     /**
-     * The list of resource that this output value depends on.
+     * The list of statically-known resources that this output value depends on. Notably, this does
+     * *not* include dependencies lifted from the result of an apply callback.
      *
      * Only callable on the outside.
      */
     /* @internal */ public readonly resources: () => Set<Resource>;
 
+    /**
+     * The list of dyanmically-known resources that this output value depends on.
+     *
+     * Only accessible on the outside.
+     */
+    // tslint:disable-next-line:variable-name
+    /* @internal */ public readonly __asyncResources: Promise<Set<Resource>> | undefined;
+
```

Comment:
I gave feedback previously about being concerned about these properties being part of the public API and therefore leading to TS-level assignability issues.  I realize now that `/* @internal */` is indeed actually being picked up by TS and these are not injected into the `.d.ts` file.  

That means that although whats implemented here is still fine, it may have been possible to make this a little cleaner.  The API doesn't need to maintain the old property here - the implementation just needs to handle the old values for non-this `Outputs` correctly.  This approach is still likely cleanest, as it keeps that split strongly typed - though it makes the implementation here rather convoluted.

Reply:
>  This approach is still likely cleanest, as it keeps that split strongly typed - though it makes the implementation here rather convoluted.

I agree.